src/test/java/org/broadinstitute/dsm/EditParticipantTest.java

```diff
@@ -0,0 +1,114 @@
+package org.broadinstitute.dsm;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.typesafe.config.Config;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.EditParticipantMessage;
+import org.broadinstitute.dsm.db.User;
+import org.broadinstitute.dsm.pubsub.EditParticipantMessagePublisher;
+import org.broadinstitute.dsm.pubsub.PubSubResultMessageSubscription;
+import org.broadinstitute.dsm.route.EditParticipantPublisherRoute;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.util.UserUtil;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class EditParticipantTest extends TestHelper {
+
+    public static final String GCP_PATH_TO_PUBSUB_PROJECT_ID = "pubsub.projectId";
+    public static final String GCP_PATH_TO_DSS_TO_DSM_SUB = "pubsub.dss_to_dsm_subscription";
+    public static final String GCP_PATH_TO_DSM_TO_DSS_TOPIC = "pubsub.dsm_to_dss_topic";
+    public static final String TEST_PAYLOAD = "{\"participantGuid\":\"TEST\",\"instanceName\":\"Angio\",\"data\":{\"firstName\":\"test\"}}";
+    public static final String UNIT_TESTER_EMAIL = "unitTesterEmail";
+
+    public Config cfg;
+    String projectId;
+    String dsmToDssSubscriptionId;
+    String topicId;
+    String messageData;
+    int userId;
+
+    @Before
+    public void first() {
+        setupDB();
+        cfg = TestHelper.cfg;
+        projectId = cfg.getString(GCP_PATH_TO_PUBSUB_PROJECT_ID);
+        dsmToDssSubscriptionId = cfg.getString(GCP_PATH_TO_DSS_TO_DSM_SUB);
+        topicId = cfg.getString(GCP_PATH_TO_DSM_TO_DSS_TOPIC);
+        messageData = TEST_PAYLOAD;
+        userId = User.getUser(cfg.getString(UNIT_TESTER_EMAIL)).getUserId();
+    }
+
+    @Test
+    public void testEditParticipantFeature() {
+
+        String realm = null;
+
+        if (UserUtil.checkUserAccess(realm, Integer.toString(userId), "mr_view") ||
+                UserUtil.checkUserAccess(realm, Integer.toString(userId), "pt_list_view")) {
```

Comment:
can you change that to the new access role `participant_edit`

Reply:
Sure, but one thing is that we will need to give permission participant_edit to unit tester user to make test pass